src/extensions/idle.rs

```diff
@@ -196,73 +191,55 @@ impl<'a, T: Read + Write + 'a> Handle<'a, T> {
             (_, result) => result,
         }
     }
-
-    /// Block until the given callback returns `false`, or until a response
-    /// arrives that is not explicitly handled by [`UnsolicitedResponse`].
-    pub fn wait_while<F>(mut self, callback: F) -> Result<()>
-    where
-        F: FnMut(UnsolicitedResponse) -> bool,
-    {
-        self.wait_inner(true, callback).map(|_| ())
-    }
 }
 
 impl<'a, T: SetReadTimeout + Read + Write + 'a> Handle<'a, T> {
-    /// Set the keep-alive interval to use when `wait_keepalive_while` is called.
+    /// Set the timeout duration on the connection. This will also set the frequency
+    /// at which the connection is refreshed.
+    ///
+    /// The interval defaults to 29 minutes as given in RFC 2177.
+    pub fn set_timeout(&mut self, interval: Duration) -> &mut Self {
+        self.timeout = interval;
+        self
+    }
+
+    /// Do not continuously refresh the IDLE connection in the background.
     ///
-    /// The interval defaults to 29 minutes as dictated by RFC 2177.
-    pub fn set_keepalive(&mut self, interval: Duration) {
-        self.keepalive = interval;
+    /// By default, connections will periodically be refreshed in the background using the
+    /// timeout duration set by [`set_timeout`]. If you do not want this behaviour, call
+    /// this function and the connection will simply IDLE until `wait_while` returns or
+    /// the timeout expires.
+    pub fn no_keepalive(&mut self) -> &mut Self {
```

Comment:
Let's make this: `fn keepalive(&mut self, enabled: bool)`.

Reply:
Done! I actually had it this way before, but changed it because it will only ever be used one way (to set the opposite value from the default), so the argument felt redundant.

Reply:
So, _technically_ there's a difference in that someone could choose to re-use a builder and want no keepalive at first but then keepalive subsequently. But yeah, it seems unlikely. Even so, I think sticking to the standard pattern is probably the way to go, if only for consistency.